hoverkite-protocol/src/response.rs

```diff
@@ -1,27 +1,26 @@
 use crate::util::{ascii_to_bool, bool_to_ascii};
-use crate::{ParseError, Side};
+#[cfg(feature = "std")]
+use crate::WriteCompat;
+use crate::{ProtocolError, Side};
+use arrayvec::ArrayString;
 use core::convert::TryInto;
 use core::mem::size_of;
 use nb::Error::{Other, WouldBlock};
-#[cfg(feature = "std")]
-use std::collections::VecDeque;
+const BUF_SIZE: usize = 256;
 
-#[cfg(feature = "std")]
-#[derive(Clone, Debug, Eq, PartialEq)]
-pub struct SideResponse {
-    pub side: Side,
-    pub response: Response,
-}
+struct TruncatingWriter(ArrayString<BUF_SIZE>);
 
-#[cfg(feature = "std")]
-#[derive(Clone, Debug, Eq, PartialEq)]
-pub enum Response {
-    Log(String),
-    Report(Report),
+impl core::fmt::Write for TruncatingWriter {
+    fn write_str(&mut self, s: &str) -> core::fmt::Result {
+        s.chars()
+            .try_for_each(|c| self.0.write_char(c))
+            .map_err(|_| core::fmt::Error)
```

Comment:
How is this different to `self.0.write_str(s)`?

Reply:
ArrayString::write_str() is atomic - it will refuse to write anything at all if it finds that the string is too long.